src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -956,7 +956,7 @@ private static void RemoveDuplicateInferredTupleNames(ArrayBuilder<string> infer
 
         private static string InferTupleElementName(ExpressionSyntax element)
         {
-            SyntaxToken nameToken = element.ExtractAnonymousTypeMemberName();
+            SyntaxToken nameToken = element.TryGetInferredMemberName();
             if (nameToken.Kind() == SyntaxKind.IdentifierToken)
```

Comment:
in all cases do we only use the nametoken if it's an identifier token?  and do we only use teh ValueText if so?  Or are there some cases where we need the actual token?  If not, i would have this API return just a string.